test/jdk/java/io/File/WindowsPrefixes.java

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright (c) 1998, 2023, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+/* @test
+ * @bug 8287843
+ * @summary Basic test for Windows path prefixes
+ * @requires (os.family == "windows")
+ * @run junit WindowsPrefixes
+ */
+
+import java.io.File;
+import java.io.IOException;
+import java.util.stream.Stream;
+
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.condition.EnabledOnOs;
+import org.junit.jupiter.api.condition.OS;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.MethodSource;
+import static org.junit.jupiter.api.Assertions.*;
+
+public class WindowsPrefixes {
+
+    private static Stream<Arguments> paths() {
+        return Stream.of(Arguments.of(""),
+                         Arguments.of("C:\\"),
+                         Arguments.of("C:"),
+                         Arguments.of("foo"),
+                         Arguments.of("foo\\bar"),
+                         Arguments.of("C:\\foo"),
+                         Arguments.of("C:foo"),
+                         Arguments.of("C:\\foo\\bar"));
```

Comment:
I suppose this should include `\foo` in the list too.

Reply:
Included in 2f31318530d7093dc69c07115c5d46b63612ff4b.